lib/routers/profile/pil.js

```diff
@@ -115,16 +116,32 @@ router.put('/:pil/:action',
   permissions('pil.update'),
   checkEstablishment,
   validateAction,
+  validate
+);
+
+router.put('/:pil/grant',
+  whitelist('procedures', 'notesCatD', 'notesCatF', 'species'),
+  updateDataAndStatus(),
+  submit('grant')
+);
+
+router.put('/:pil/revoke',
+  whitelist('comments'),
+  updateDataAndStatus(),
+  submit('revoke')
+);
+
+router.put('/:pil/transfer',
+  whitelist('procedures', 'notesCatD', 'notesCatF', 'species', 'establishment'),
+  updateDataAndStatus(),
   (req, res, next) => {
-    const action = req.params.action;
-    if (action === 'revoke') {
-      return whitelist('comments')(req, res, next);
+    req.establishment.id = get(req.body.data, 'establishment.to.id');
+    if (!req.user.profile.establishments.find(e => e.id === req.establishment.id)) {
```

Comment:
req.user is the logged in user, req.profile is the scoped user - it feels like we should use the scoped user else we are hard-baking business logic in

Reply:
`req.profile.establishments` is scoped to the current establishment (`req.establishment`) in the api so we can't see any establishments other than the one the pil is currently at. A user can only transfer their own PIL so this should be fine.